SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialLoadBalancingHandler.cpp

```diff
@@ -46,6 +47,45 @@ FSpatialLoadBalancingHandler::EvaluateActorResult FSpatialLoadBalancingHandler::
 	if (NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID))
 	{
 		AActor* NetOwner = GetReplicatedHierarchyRoot(Actor);
+
+		if (AController* Controller = Cast<AController>(Actor))
+		{
+			TArray<UActorComponent*> Components;
+			Controller->GetComponents(URemotePossessionComponent::StaticClass(), Components);
+			if (Components.Num() == 1)
+			{
+				if (URemotePossessionComponent* Component = Cast<URemotePossessionComponent>(Components[0]))
+				{
+					if (NetDriver->LockingPolicy->IsLocked(Actor))
+					{
+						UE_LOG(LogSpatialLoadBalancingHandler, Warning, TEXT("Actor %s (%llu) cannot migrate because it is locked"),
+							   *Actor->GetName(), EntityId);
+						Component->DestroyComponent();
+						return EvaluateActorResult::None;
+					}
+
+					if (Component->Target->HasAuthority())
```

Comment:
Ok, I see, that is where the local possession is handled.
One thing that is a bit risky about doing it in the middle of the NetDriver loop is that when starting, we have a list of actors, with properties values, and relationships between actors. Doing the possess in the middle of the driver's loop might change these and cause determinism issues.
As in, depending the order in which you evaluate the actor array, you read actors A,B,C,D, with B possessing A, and took some decision regarding load balancing. Now you evaluate C, which evaluates the RemotePossessionComponent, and Decides to possess A. Now, some of the decisions we have taken are wrong.

We should schedule this so it happens in the game loop (in Tick, if there is not better place).

Reply:
From my understanding, you worry about the order for calling EvaluateSingleActor, maybe the possession requests coming as A, B, C, D, and C wants to Possess A, B wants to Possess A. 
But the order of Actors from `iCtx.GetActorsBeingReplicated()` is not to be exactly the same as the order of possession requests. Is it right?

I had an experiment about OnAuthorityGained order as [this ticket](https://improbableio.atlassian.net/browse/UNR-4739)
It didn't have a certain order for `OnAuthorityGained`. 

So, the decisions in load balancing are not important. 
As I understand it, which Controller will successfully take possession of Pawn depends on who calls Possess first, not the order of the requests.
And to really make it happen, whoever requests it first may need the OnAuthorityGained call to be FIFO as well.

Reply:
This is not about OnAuthorityGained. This callback happens on checkout on another server, and order could be whatever the SpatialOS runtime decides.
This is about the LoadBalancing taking place into TickFlush, which takes the list of actors to replicate, with a state set by the user, and sends it to the network. This step is not supposed to change the actors themselves